evalml/model_understanding/graphs.py

```diff
@@ -501,6 +501,46 @@ def graph_binary_objective_vs_threshold(pipeline, X, y, objective, steps=100):
     return _go.Figure(layout=layout, data=data)
 
 
+def _get_feature_names_from_str_or_col_index(X, names_or_col_indices):
+    """Helper function to map the user-input features param to column names."""
+    feature_list = []
+    for name_or_index in names_or_col_indices:
+        if isinstance(name_or_index, int):
+            feature_list.append(X.columns[name_or_index])
+        else:
+            feature_list.append(name_or_index)
+    return feature_list
+
+
+def _raise_value_error_if_any_features_all_nan(df):
+    """Helper for partial dependence data validation."""
+
+    nan_pct = df.isna().mean()
+    all_nan = nan_pct[nan_pct == 1].index.tolist()
+    all_nan = [f"'{name}'" for name in all_nan]
+
+    if all_nan:
+        raise ValueError("The following features have all NaN values and so the "
+                         f"partial dependence cannot be computed: {', '.join(all_nan)}")
+
+
+def _raise_value_error_if_mostly_one_value(df, percentile):
```

Comment:
This is to fix #2119. May be scope-creep but it felt weird to have the all-NaN check work for two-way without fixing this warning. The implementation is pretty simple, we basically use the existing code for the one-way check on every column the user passed in, so I thought it was appropriate.

Reply:
Is it weird not using our own datachecks to accomplish this type of thing?  This seems very much like a uniqueness and a no variance datacheck.

Reply:
Good suggestion! There's definitely overlap in the logic. I'll file a spike issue.